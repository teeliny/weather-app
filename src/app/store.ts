import { configureStore } from '@reduxjs/toolkit';

import {  apiSlice } from '../features/weather/weather-api-slice';
import screenReducer from '../features/screen/screen-width-slice';

export const store = configureStore({
  reducer: {
    screen: screenReducer,
    [apiSlice.reducerPath]: apiSlice.reducer
  },
  middleware: (getDefaultMiddleware) => {
    return getDefaultMiddleware().concat(apiSlice.middleware);
  }
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
