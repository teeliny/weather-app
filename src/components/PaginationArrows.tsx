import React from 'react';
import Container from '@mui/material/Container';
import Button from '@mui/material/Button';
import ArrowCircleRightIcon from '@mui/icons-material/ArrowCircleRight';
import ArrowCircleLeftIcon from '@mui/icons-material/ArrowCircleLeft';

interface IPaginationArrow {
  handleLeft: () => void;
  handleRight: () => void;
  handleRefetch: () => void;
  pageIndex: number;
  maxPage: number;
}

function PaginationArrows({ handleLeft, handleRight, handleRefetch, pageIndex, maxPage }: IPaginationArrow) {
  
  return (
    <Container
      sx={{
        minWidth: '100%',
        display: 'flex',
        justifyContent: 'space-between',
        alignItems: 'center',
        paddingLeft: 0,
        paddingRight: 0,
        margin: 0,
      }}
    >
      {
        <ArrowCircleLeftIcon
          sx={{ fontSize: 36.5, opacity: pageIndex === 1 ? '0.1' : '1' }}
          onClick={handleLeft}
        />
      }
      <Button
        sx={{
          textTransform: 'capitalize',
          backgroundColor: '#ffffff',
          border: '1px solid #000000',
          color: '#000000',
        }}
        onClick={handleRefetch}
      >
        Refresh
      </Button>
      {
        <ArrowCircleRightIcon
          sx={{ fontSize: 36.5, opacity: pageIndex === maxPage ? '0.1' : '1' }}
          onClick={handleRight}
        />
      }
    </Container>
  );
}

export default PaginationArrows;

