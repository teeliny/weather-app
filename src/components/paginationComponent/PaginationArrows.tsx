import React from 'react';
import Container from '@mui/material/Container';
import Button from '@mui/material/Button';
import ArrowCircleRightIcon from '@mui/icons-material/ArrowCircleRight';
import ArrowCircleLeftIcon from '@mui/icons-material/ArrowCircleLeft';
import { useAppSelector } from '../../app/hooks';
import { IPaginationArrow } from './pagination.typing';

function PaginationArrows({ handleLeft, handleRight, handleRefetch, pageIndex, maxPage }: IPaginationArrow) {
  const myView = useAppSelector((state) => state.screen.mobile_view);
  return (
    <Container
      sx={{
        position: 'absolute',
        top: myView ? '' : '175px',
        minWidth: '100%',
        display: 'flex',
        justifyContent: 'space-between',
        alignItems: 'center',
        paddingLeft: '0',
        paddingRight: '0',
        margin: '0',
      }}
    >
      {
        <ArrowCircleLeftIcon
          sx={{ fontSize: 36.5, opacity: pageIndex === 1 ? '0.1' : '1' }}
          onClick={handleLeft}
        />
      }
      {myView && (
        <Button
          sx={{
            textTransform: 'capitalize',
            backgroundColor: '#ffffff',
            border: '1px solid #000000',
            color: '#000000',
            width: '80px',
          }}
          onClick={handleRefetch}
        >
          Refresh
        </Button>
      )}
      {
        <ArrowCircleRightIcon
          sx={{ fontSize: 36.5, opacity: pageIndex === maxPage ? '0.1' : '1' }}
          onClick={handleRight}
        />
      }
    </Container>
  );
}

export default PaginationArrows;

