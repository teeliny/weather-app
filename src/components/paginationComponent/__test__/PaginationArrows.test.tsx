import React from 'react';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { store } from '../../../app/store';
import PaginationArrows from '../PaginationArrows';

let eleWrapper: JSX.Element;

beforeEach(() => {
  eleWrapper = (
    <Provider store={store}>
      <PaginationArrows
        handleLeft={function (): void {
          throw new Error('Function not implemented.');
        }}
        handleRight={function (): void {
          throw new Error('Function not implemented.');
        }}
        handleRefetch={function (): void {
          throw new Error('Function not implemented.');
        }}
        pageIndex={0}
        maxPage={0}
      />
    </Provider>
  )
});
afterAll(cleanup);

it('render pagination arrows correctly', () => {
  const { getByTestId } = render(eleWrapper);
  expect(getByTestId('ArrowCircleLeftIcon')).toBeTruthy();
  expect(getByTestId('ArrowCircleRightIcon')).toBeTruthy();
});

it('matches pagination arrow', () => {
  const tree = renderer.create(eleWrapper).toJSON();
  expect(tree).toMatchSnapshot();
});