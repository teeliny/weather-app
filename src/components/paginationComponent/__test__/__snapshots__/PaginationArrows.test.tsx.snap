// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches pagination arrow 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthLg css-yu64m-MuiContainer-root"
>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1k0tyso-MuiSvgIcon-root"
    data-testid="ArrowCircleLeftIcon"
    focusable="false"
    onClick={[Function]}
    viewBox="0 0 24 24"
  >
    <path
      d="M2 12c0 5.52 4.48 10 10 10s10-4.48 10-10S17.52 2 12 2 2 6.48 2 12zm10-1h4v2h-4v3l-4-4 4-4v3z"
    />
  </svg>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-z9fw94-MuiSvgIcon-root"
    data-testid="ArrowCircleRightIcon"
    focusable="false"
    onClick={[Function]}
    viewBox="0 0 24 24"
  >
    <path
      d="M22 12c0-5.52-4.48-10-10-10S2 6.48 2 12s4.48 10 10 10 10-4.48 10-10zm-10 1H8v-2h4V8l4 4-4 4v-3z"
    />
  </svg>
</div>
`;
